pkgs/development/python-modules/pytest-mpl/default.nix

```diff
@@ -43,10 +43,13 @@ buildPythonPackage rec {
     "tests/subtests/test_subtest.py"
   ];
 
+  # need to set MPLBACKEND=agg for headless matplotlib for darwin
+  # https://github.com/matplotlib/matplotlib/issues/26292
   # The default tolerance is too strict in our build environment
   # https://github.com/matplotlib/pytest-mpl/pull/9
   # https://github.com/matplotlib/pytest-mpl/issues/225
   preCheck = ''
+    export MPLBACKEND=agg
```

Comment:
this seems like it could get set in a setup hook so consumers of package also have this set

Reply:
setupHook could do weird things in dev shells -- as noted in matrix by Emily